### DISCUSSION TITLE
VaporFile Download Redirect

### DISCUSSION THREAD

**Discussion by lonnylot** [laravel/nova-issues#5604] [Ideas & Feature Requests] [answered]
VaporFile is meant to handle file uploads >6MB.

However, by default, when downloading the file you pipe the stream through Vapor even though Lambda / Vapor can only handle files < 6MB.

I'd like to propose changing the default download callback to return a redirect to a temporary URL.

This has the added benefit that if the file is binary you don't have to set a custom header https://docs.vapor.build/1.0/projects/development.html#binary-responses

**Comment by crynobone** [comment]
Shouldn't you be able to do the following?

```php
use Illuminate\Support\Facades\Storage;
use Laravel\Nova\Fields\VaporImage;

VaporImage::make('Photo', 'photo')
    ->download(function ($request, $model, $disk, $value) {
        return redirect(Storage::disk($disk)->temporaryUrl(
            $value, now()->addMinutes(5)
        ));
    }),
```

**Comment by lonnylot** [comment]
@crynobone 
Yeah, but I'm suggesting something like that should be the default. If the whole point of VaporFile is to handle files > 6MB on upload then I'd think by default it would handle files > 6MB on download. Or at the very least the docs should be updated to share that even though it works on upload, it won't work on download.

**Reply by crynobone** [reply, marked as answer]
This has been added to our documentation: https://nova.laravel.com/docs/4.0/resources/file-fields.html#downloading-large-files-with-temporary-urls
